src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1521,6 +1527,81 @@ public bool MoveNext()
             return true;
         }
     }
+
+    /// <summary>
+    /// The simple enumerator class is used for the range operator '..'
+    /// in expressions like 'A'..'B' | ForEach-Object { $_ }
+    /// </summary>
+    internal class CharRangeEnumerator : IEnumerator
```

Comment:
I'm curious if we should start following the one-type-per-file rule https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1402.md

Reply:
It is often useful to have related types in the same file, so I'm strongly opposed to following this rule. I'm strongly in favor of organizing code in ways that make sense to the author and reviewers. We shouldn't blindly follow the guidelines of some tool.

Reply:
I agree that it make sense to group related code and that following the rules blindly is silly. But, some of the files in this project are so massive that there is 0 benefit gained by the code being grouped, IMO. If I need to have 3 tabs open on the same file to work on it effectively, the code is probably better split, IMO. Doesn't have to be strictly followed, but, maybe give some serious consideration?

Reply:
Are you suggesting the author of said code saw 0 benefit of a huge file, because you might be saying that to said author. (I'm the primary author of 4 of the 10 largest non-generated files in the repo.)

To pick on the largest file - Ast.cs - there are at least 72 classes in that file. When I know I'm looking for something Ast related, but not sure what, if I end up with 72 open tabs that I need to close, I'm less productive than if I open 1 file.

And if you needed 3 tabs open to effectively work on some code, you needed 3 tabs to see some code, 1 file or 3 files, right? I do realize 

There is always room for improvement, but it should be thoughtful and not rely on some tool - tools are good at enforcing things that are easy to check, less good at providing good judgement.

Reply:
>Are you suggesting the author of said code saw 0 benefit of a huge file,

Not at all. I'm suggesting *I* see 0 benefit.

> because you might be saying that to said author.

I know 🙂 .

> And if you needed 3 tabs open to effectively work on some code, you needed 3 tabs to see some code, 1 file or 3 files, right?

Right, I'm saying I see no benefit to these being in the same file, where I do see benefit in it being in separate files. I think it's a difference of preference, I prefer smaller files grouped in folders over monolith files. It's easier to jump between code in different files using the IDE, IMO. When the code is in the same file, the default action is to jump to the code in the current tab. When it is in another file, a new tab opens with that file and my current place in the previous file is preserved. *shrugs

I'm also trying to approach this from a maintainable open source project perspective. If this rule doen't makes sense for this project, then it doesn't make sense and we don't use it. I just want to have something in place that can help maintain and encourage an enforceable style for the life of the project, preferably through CI. It can help shift the burden from reviewers to contributors. 

If I'm wasting my efforts there, then I will gladly redirect my attention elsewhere.